modeltrans/forms.py

```diff
@@ -0,0 +1,348 @@
+                if original_field in new_field_order:
+                    index = new_field_order.index(original_field) + 1
+                    new_field_order[index:index] = self.included_fields[original_field]
+                    new_field_order.pop(index - 1)
+            field_order = new_field_order
+        else:
+            # if no explicit field order adopt the order of i18n form fields used to generate the included fields
+            field_order = self.included_fields["__all__"]
+
+        self.order_fields(field_order)
+
+    def remove_excess_fields(self):
+        """Remove translations fields that are not included in languages."""
+        fields = self.fields.copy()
+        for original_field_name in self.i18n_fields:
+            for field in fields:
+                if original_field_name in field and field not in self.included_fields["__all__"]:
+                    self.fields.pop(field)
+
+    def get_fallback_language(self, fallback_language=None):
+        """
+        Get the fallback language.
+
+        Priority is defined as:
+        1) key word argument
+        2) form Meta option
+        3) model instance custom fallback
+        4) system default fallback
+        """
+        if fallback_language:
+            return fallback_language
+
+        if self._meta.fallback_language:
+            return self._meta.fallback_language
+
+        if self.model_i18n_field.fallback_language_field and self.instance.pk:
+            return get_instance_field_value(
+                self.instance, self.model_i18n_field.fallback_language_field
+            )
+
+        return get_default_language()
+
+    def get_language_codes(self):
+        """Get the languages based on options included in include languages, in the order they are submitted."""
+
+        languages = []
+        if self.languages:
+
+            for value in self.languages:
+                if not isinstance(value, str):
+                    raise ValueError("languages: values should be strings")
+                # TODO: need to consider what to do with en_GB style languages ...?
+                if len(value) > 2 and value not in LANGUAGE_OPTIONS:
+                    raise ValueError(f"languages: value {value} is not permitted")
+                if len(value) < 2:
+                    raise ValueError(f"languages: value {value} is not permitted")
+                if len(value) == 2 and value not in get_available_languages():
+                    raise ValueError(
+                        f"languages: {value} is not an available language in the system"
+                    )
+
+                if value == "browser":
+                    languages.append(get_language())
+                elif value == "fallback":
+                    languages.append(self.fallback_language)
+                else:
+                    languages.append(value)
+
+        if not languages:
+            raise ValueError("languages: Error. No languages have been defined.")
```

Comment:
Error is implied by the exception type?

```suggestion
            raise ValueError("languages: No languages have been defined.")
```